saas/models/saas_template.py

```diff
@@ -119,6 +120,12 @@ class SAASTemplateLine(models.Model):
 
     ], default='draft')
 
+    @api.constrains('operator_db_name')
+    def _check_db(self):
+        if self.operator_db_name in db.list_dbs():
+            raise ValidationError(_('Database name must be unique'))
```

Comment:
Actually :-1: 
Because it will work only in "same instance" operator and doesn't make sense in multi-server environment. What you can do it is check uniqueness across saas.db and saas.template.operator, though there is no easy way to do cross-table uniqueness

Reply:
As a workaround, we can add a prefix for database name, e.g. ``$` and then add constrain to template and to normal databases:

* template database: must start with $, must me unique across template database names. (Note, that we can add prefix ``$`` automatically 
* build database: cannot start with $, must be unique across database name